### ISSUE TITLE
gradient checkpointing runs during validations in the training examples

### ISSUE THREAD

**Issue by bghira** [huggingface/diffusers#10107]
### Describe the bug

The gradient checkpointing is enabled via self.training, but then the log_validations also unnecessarily encounter this codepath.

I found that I have to disable this when running even under `torch.no_grad()`, looked and saw that the official examples do not do this either.

This gives a substantial performance boost more similar to a normal inference script running outside of a training loop.

### Reproduction

Add print statements to the checkpointing function.

### System Info

-

### Who can help?

@linoytsaban @yiyixuxu

**Comment by bghira** [comment]
this impacted every model I looked at while implementing layer-skipping for checkpointing.

**Comment by bghira** [comment]
@a-r-r-o-w see this one as well

**Comment by yiyixuxu** [comment]
we have this PR recently https://github.com/huggingface/diffusers/pull/9878
I think it is gradient checkpoint won't be anbled if you have `torch.no_grads()` now
